src/doc/book/documentation.md

```diff
@@ -460,8 +460,9 @@ not actually pass as a test.
 ```
 
 The `no_run` attribute will compile your code, but not run it. This is
-important for examples such as "Here's how to start up a network service,"
-which you would want to make sure compile, but might run in an infinite loop!
+important for examples such as "Here's how to retrive a web page,"
+which you would want to ensure it compiles, but might be run in a test
```

Comment:
the original grammar is odd but correct, since "examples" is plural and so the sentence is "examples [..], which you would want to ensure compile." Adding "it" is incorrect because "which" is already the subject of "compile". Alternatively you could write e.g. "an example such as [..]: you would want to ensure it compiles, but it might be..." which is probably less confusing overall.